### DISCUSSION TITLE
Minio can't keep empty folders?

### DISCUSSION THREAD

**Discussion by xigua88-66** [minio/minio#15146] [Q&A] [answered]
### Discussed in https://github.com/minio/minio/discussions/15139

<div type='discussions-op-text'>

<sup>Originally posted by **xigua88-66** June 22, 2022</sup>
I use unzip to extract a zip package to the Minio bucket, if the zip contains an empty folder, the folder will not be created.
Likewise, if I use s3cmd for put and cp, if the source directory has a blank folder, the destination path doesn't generate that folder</div>

**Comment by klauspost** [comment, marked as answer]
There is no concept of "folders" in S3. You have objects with keys. Keys can have shared prefixes that behave like folders, but they are prefixes tied to objects. Therefore no empty "prefix" can exist.

You can create empty objects ending with a `/` to make it seem like there is a folder, but it is generally an antipattern.
